blackjax/mcmc/mala.py

```diff
@@ -0,0 +1,63 @@
+"""Public API for Metropolis Adjusted Langevin kernels."""
+from typing import Callable, NamedTuple, Tuple
+
+import jax
+import jax.numpy as jnp
+
+from blackjax.mcmc.diffusion import overdamped_langevin
+from blackjax.types import PRNGKey, PyTree
+
+__all__ = ["MALAState", "MALAInfo", "init", "kernel"]
+
+
+class MALAState(NamedTuple):
+    position: PyTree
+    logprob_grad: PyTree
+
+
+class MALAInfo(NamedTuple):
+    acceptance_probability: float
+    is_accepted: bool
+
+
+def init(position: PyTree, logprob_fn: Callable) -> MALAState:
+    grad_fn = jax.grad(logprob_fn)
+    logprob_grad = grad_fn(position)
+    return MALAState(position, logprob_grad)
+
+
+def kernel(logprob_fn: Callable, step_size: float):
+    grad_fn = jax.grad(logprob_fn)
+    integrator = overdamped_langevin(grad_fn)
+
+    def transition_probability(state, new_state, step_size):
+        """Transition probability to go from `state` to `new_state`"""
+        theta = new_state.position - state.position - step_size * state.logprob_grad
+        return -0.25 * (1.0 / step_size) * jnp.dot(theta, theta)
+
+    def one_step(rng_key: PRNGKey, state: MALAState) -> Tuple[MALAState, MALAInfo]:
+
+        new_state = integrator(rng_key, state, step_size)
+
+        delta = (
+            logprob_fn(new_state.position)
```

Comment:
Ideally, the log-probability is calculated in the `integrator` call above alongside the gradient (as it would be cheaper computationally to calculate both in one go). This would mean adding `log_probability` to `DiffusionState`, and well as using `value_and_grad` in `overdamped_langevin`.

Do you think this should be done now or as an imrovement later on?

Reply:
Yep that's a good idea, just made the change.